plenty_api/api.py

```diff
@@ -518,43 +524,85 @@ def __repeat_get_request_for_all_unpaginated_records(self,
         page_info = utils.sniff_response_format(response=response)
         entries = response[page_info['data']]
 
-        while True:
-            logging.debug(f"fetching page {query['page'] + 1}")
-            query.update({'page': query['page'] + 1})
-            response = self.__plenty_api_request(method='get',
-                                                 domain=domain,
-                                                 path=path,
-                                                 query=query)
-            if not response:
-                return None
+        if len(response[page_info['data']]) == query['itemsPerPage']:
+            while True:
+                logging.debug(f"fetching page {query['page'] + 1}")
+                query.update({'page': query['page'] + 1})
+                response = self.__plenty_api_request(method='get',
+                                                    domain=domain,
+                                                    path=path,
+                                                    query=query)
+                if not response:
+                    return None
 
-            if isinstance(response, dict) and 'error' in response.keys():
-                logging.error(f"subsequent {domain} API requests failed.")
-                return response
+                if isinstance(response, dict) and 'error' in response.keys():
+                    logging.error(f"subsequent {domain} API requests failed.")
+                    return response
 
-            entries += response[page_info['data']]
+                entries += response[page_info['data']]
 
-            if len(response[page_info['data']]) < query['itemsPerPage']:
-                break
+                if len(response[page_info['data']]) < query['itemsPerPage']:
+                    break
 
         return entries
 
 
+    def __get_request_dump_file(self,
+                                             domain: str,
+                                             query: dict,
+                                             path: str = '',
+                                             download_filepath: str = '') -> dict:
```

Comment:
Domain and path are not needed as there is only a single user at this point, we can generalize once we get a second user of that function. Just use domain = 'bi_raw' and path = '/file' statically.